8290368: Introduce LDAP and RMI protocol-specific object factory filters to JNDI implementation

Pooja-DP opened the pull request:
This pull request contains a backport of commit https://github.com/openjdk/jdk17u/commit/39a6bf2158c2f35399c6739e4b9578f13ea6b7e4 from the https://github.com/openjdk/jdk17u

Bug Ref: https://bugs.openjdk.org/browse/JDK-8290368

Signed-off-by: Pooja.D.P1 <Pooja.D.P1@ibm.com>

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- \[ \] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- \[x\] Change must not contain extraneous whitespace
- \[x\] Commit message must refer to an issue
- \[ \] [JDK-8290368](https://bugs.openjdk.org/browse/JDK-8290368) needs maintainer approval
- \[x\] Change requires CSR request [JDK-8360532](https://bugs.openjdk.org/browse/JDK-8360532) to be approved

### Error
&nbsp;⚠️ Pull request body is missing required line: `- [x] I confirm that I make this contribution in accordance with the [OpenJDK Interim AI Policy](https://openjdk.org/legal/ai).`

### Issues
 * [JDK-8290368](https://bugs.openjdk.org/browse/JDK-8290368): Introduce LDAP and RMI protocol-specific object factory filters to JNDI implementation (**Bug** - P3)
 * [JDK-8360532](https://bugs.openjdk.org/browse/JDK-8360532): Introduce LDAP and RMI protocol-specific object factory filters to JNDI implementation (**CSR**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk11u-dev.git pull/3111/head:pull/3111` \
`$ git checkout pull/3111`

Update a local copy of the PR: \
`$ git checkout pull/3111` \
`$ git pull https://git.openjdk.org/jdk11u-dev.git pull/3111/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3111`

View PR using the GUI difftool: \
`$ git pr show -t 3111`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk11u-dev/pull/3111.diff">https://git.openjdk.org/jdk11u-dev/pull/3111.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk11u-dev/pull/3111#issuecomment-3480328994)
</details>

Pooja-DP committed:
Backport 39a6bf2158c2f35399c6739e4b9578f13ea6b7e4

Signed-off-by: Pooja.D.P <Pooja.D.P1@ibm.com>

Pooja-DP commented:
@RealCLanger - Could you please review my PR?

gnu-andrew reviewed (changes requested):
It seems that when 8244473 was backported to 11u, documentation changes were not included so I think it is correct to include them here, along with fixing the ordering in `java.security` caused by patches being applied in a different order as far as I can see.

There are still some conflict hunks present in `src/java.naming/share/classes/module-info.java` and a couple of bits from 8244473 still missing:

~~~
--- src/java.naming/share/classes/module-info.java      2025-12-03 09:33:23.798486609 +0000
+++ ../jdk/src/java.naming/share/classes/module-info.java       2025-12-03 01:34:51.274570322 +0000
@@ -29,11 +29,11 @@
  * Common standard JNDI environment properties that may be supported
  * by JNDI providers are defined and documented in
  * {@link javax.naming.Context}. Specific JNDI provider implementations
- * may also support other environment properties, which are specific
+ * may also support other environment or system properties, which are specific
  * to their implementation.
  *
  * @implNote
- * The following implementation specific properties are supported by the
+ * The following implementation specific environment properties are supported by the
  * default LDAP Naming Service Provider implementation in the JDK:
  * <ul>
  *     <li>{@code java.naming.ldap.factory.socket}:
@@ -86,8 +86,6 @@
  *         channel binding information to the server.
  *     </li>
  * </ul>
-<<<<<<< HEAD
-=======
  * <p>The following implementation specific system properties are supported by the
  * default LDAP Naming Service Provider implementation in the JDK:
  * <ul>
~~~

If you can fix this up, this should be ready to go.

What testing has been done with this change?

Pooja-DP commented:
/keepalive

Pooja-DP commented:
@gnu-andrew : Thanks for reviewing my PR. I have addressed your comments and enabled git action as well https://github.com/Pooja-DP/jdk11u-dev/actions/runs/20845054449  Please take a look. Thanks

Pooja-DP commented:
/touch
